binr/r2r/r2r.c

```diff
@@ -853,20 +853,33 @@ static void fixup_tests(RPVector *results, const char *edited_file, ut64 start_l
 	}
 }
 
-static void replace_cmd_kv(const char *path, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
-	char *content = r_file_slurp (path, NULL);
-	if (!content) {
-		eprintf ("Failed to read file \"%s\"\n", path);
-		return;
-	}
+static char *replace_cmd_kv(const char *path, const char *content, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
 	char *kv = format_cmd_kv (key, value);
 	if (!kv) {
-		free (content);
-		return;
+		return NULL;
 	}
-	ut64 kv_lines = r_str_char_count (kv, '\n');
+	ut64 kv_lines = r_str_char_count (kv, '\n') + 1;
 	char *newc = replace_lines (content, line_begin, line_end, kv);
 	free (kv);
+	if (!newc) {
+		return NULL;
+	}
+	ut64 lines_before = line_end - line_begin;
+	st64 delta = (st64)kv_lines - lines_before;
```

Comment:
i dont think delta can be negative, so make this signed and chk if kv_lines < lines_before and return early in that case

Reply:
Yes, delta can be negative, for example if you fix from:
```
EXPECT=<<EOF
a
lot
of
broken
stuff
EOF
```
to
```
EXPECT=<<EOF
correct stuff
EOF
```

Reply:
ok